ejb/src/main/java/org/wildfly/httpclient/ejb/ServerHandlers.java

```diff
@@ -0,0 +1,756 @@
+                    }
+
+    //                @Override
+                    public SecurityIdentity getSecurityIdentity() {
+                        return exchange.getAttachment(ElytronIdentityHandler.IDENTITY_KEY);
+                    }
+
+                    @Override
+                    public void writeException(@NotNull Exception exception) {
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.INTERNAL_SERVER_ERROR, exception);
+                    }
+
+                    @Override
+                    public void writeNoSuchEJB() {
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.NOT_FOUND, new NoSuchEJBException());
+                    }
+
+                    @Override
+                    public void writeWrongViewType() {
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.NOT_FOUND, EjbHttpClientMessages.MESSAGES.wrongViewType());
+                    }
+
+                    @Override
+                    public void writeCancelResponse() {
+                        throw new RuntimeException("nyi");
+                    }
+
+                    @Override
+                    public void writeNotStateful() {
+                        HttpServerHelper.sendException(exchange, httpServiceConfig, StatusCodes.INTERNAL_SERVER_ERROR, EjbHttpClientMessages.MESSAGES.notStateful());
+                    }
+
+                    @Override
+                    public void convertToStateful(@NotNull SessionID sessionId) throws IllegalArgumentException, IllegalStateException {
+
+                        Cookie sessionCookie = exchange.getRequestCookie(JSESSIONID_COOKIE_NAME);
+                        if (sessionCookie == null) {
+                            String rootPath = exchange.getResolvedPath();
+                            int ejbIndex = rootPath.lastIndexOf("/ejb");
+                            if (ejbIndex > 0) {
+                                rootPath = rootPath.substring(0, ejbIndex);
+                            }
+                            exchange.setResponseCookie(new CookieImpl(JSESSIONID_COOKIE_NAME, sessionIdGenerator.createSessionId()).setPath(rootPath));
+                        }
+
+                        exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, EJB_RESPONSE_NEW_SESSION.toString());
+                        exchange.getResponseHeaders().put(EJB_SESSION_ID, Base64.getUrlEncoder().encodeToString(sessionId.getEncodedForm()));
+
+                        exchange.setStatusCode(StatusCodes.NO_CONTENT);
+                        exchange.endExchange();
+                    }
+
+                    @Override
+                    public <C> C getProviderInterface(Class<C> providerInterfaceType) {
+                        return null;
+                    }
+                });
+            });
+        }
+
+        private static String handleDash(String s) {
```

Comment:
Move handleDash to RemoteHTTPHandler as a common method.

Reply:
Done.